### PR TITLE
misc: add github action for running go test

### PR DESCRIPTION
**Pull request by worm-emoji**
this pull request adds a github workflow that runs `go test` for the repo. nobody asked me to do this but i decided to do it anyway.

### CODE REVIEW

**Comment by worm-emoji** [comment]
not sure if i have the right permissions on this repo so that this will actually run, but yeah. hopefully this works!

File: .github/workflows/go-test.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.20"
+      - uses: actions/cache@v3
+        with:
```
Comment (ryandotsmith) [inline .github/workflows/go-test.yml:20]: I'm curious what motived this with paragraph. I didn't see it in the official doc: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go

**Review by ryandotsmith — COMMENTED** [review]
amazing! ty!

File: .github/workflows/go-test.yml (hunk 1 above)
Comment (worm-emoji) [reply .github/workflows/go-test.yml:20]: Idea is to cache the go mod folders for faster incremental builds. Happy to omit it

Comment (worm-emoji) [reply .github/workflows/go-test.yml:20]: Docs here: https://github.com/actions/cache/blob/main/examples.md#linux-1

Comment (ryandotsmith) [reply .github/workflows/go-test.yml:19]: wonderful. can you add that link as a comment in the file?

Comment (worm-emoji) [reply .github/workflows/go-test.yml:21]: done!

**Review by ryandotsmith — APPROVED** [review]

**Pull request merged**